jupyter/d2l-java/chapter_convolutional-neural-networks/lenet.ipynb

```diff
@@ -0,0 +1,577 @@
+    "import ai.djl.training.initializer.XavierInitializer;\n",
+    "import ai.djl.training.listener.TrainingListener;\n",
+    "import ai.djl.training.loss.Loss;\n",
+    "import ai.djl.training.optimizer.Optimizer;\n",
+    "import ai.djl.training.optimizer.learningrate.LearningRateTracker;\n",
+    "import tech.tablesaw.api.*;\n",
+    "import tech.tablesaw.plotly.api.*;\n",
+    "import tech.tablesaw.plotly.components.*;\n",
+    "import tech.tablesaw.plotly.Plot;\n",
+    "import tech.tablesaw.plotly.components.Figure;\n",
+    "import org.apache.commons.lang3.ArrayUtils;"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "NDManager manager = NDManager.newBaseManager();\n",
+    "\n",
+    "SequentialBlock block = new SequentialBlock();\n",
+    "\n",
+    "block.add(\n",
+    "    Conv2D.builder()\n",
+    "            .setKernel(new Shape(5, 5))\n",
+    "            .optPad(new Shape(2, 2))\n",
+    "            .optBias(false)\n",
+    "            .setNumFilters(6).build())\n",
+    ".add(Activation::sigmoid)\n",
+    ".add(\n",
+    "    Pool\n",
+    "    .avgPool2DBlock(new Shape(5, 5), new Shape(2, 2), new Shape(2, 2)))\n",
+    ".add(\n",
+    "    Conv2D.builder()\n",
+    "            .setKernel(new Shape(5, 5))\n",
+    "            .setNumFilters(16).build())\n",
+    "    \n",
+    ".add(Activation::sigmoid)   \n",
+    ".add(\n",
+    "    Pool\n",
+    "    .avgPool2DBlock(new Shape(5, 5), new Shape(2, 2), new Shape(2, 2)))\n",
+    ".add(\n",
+    "    Linear.builder()\n",
+    "    .optFlatten(true)\n",
+    "    .setOutChannels(120)\n",
+    "    .build())\n",
+    ".add(Activation::sigmoid)\n",
+    ".add(\n",
+    "    Linear.builder()\n",
+    "    .setOutChannels(84)\n",
+    "    .build())\n",
+    ".add(Activation::sigmoid)\n",
+    ".add(\n",
+    "    Linear\n",
+    "    .builder()\n",
+    "    .setOutChannels(10)\n",
+    "    .build());\n",
+    "\n",
+    "block.setInitializer(new XavierInitializer());"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We took a small liberty with the original model,\n",
+    "removing the Gaussian activation in the final layer.\n",
+    "Other than that, this network matches\n",
+    "the original LeNet5 architecture.\n",
+    "\n",
+    "By passing a single-channel (black and white)\n",
+    "$28 \\times 28$ image through the net\n",
+    "and printing the output shape at each layer,\n",
+    "we can inspect the model to make sure \n",
+    "that its operations line up with \n",
+    "what we expect from :numref:`img_lenet_vert`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "NDArray X = manager.randomUniform(0f, 1.0f, new Shape(1, 1, 28, 28));\n",
+    "block.initialize(manager, DataType.FLOAT32, X.getShape());\n",
```

Comment:
Can you use getOutputShapes() in this section instead of forward? It would mean that you can skip the initialization and move it down with the training